homeassistant/components/rfxtrx.py

```diff
@@ -31,6 +38,42 @@
 _LOGGER = logging.getLogger(__name__)
 RFXOBJECT = None
 
+DEVICE_SCHEMA = vol.Schema({
+    vol.Required(ATTR_NAME): cv.string,
+    vol.Required(ATTR_PACKETID): cv.slug,
+    vol.Optional(ATTR_FIREEVENT, default=False): cv.boolean,
+})
+
+
+def _valid_device(value):
+    """Validate a dictionary of devices definitions."""
+    config = OrderedDict()
+    for key, device in value.items():
+        try:
+            key = cv.slug(key)
+            config[key] = DEVICE_SCHEMA(device)
+        except vol.MultipleInvalid as ex:
+            raise vol.Invalid('Rfxtrx device {} is invalid: {}'
+                              .format(key, ex))
+    return config
+
+
+DEFAULT_SCHEMA = vol.Schema({
+    vol.Required("platform"): DOMAIN,
+    vol.Required(CONF_DEVICES): vol.All(dict, _valid_device),
```

Comment:
I know you copied this from the group component but it's wrong :-P Or, not wrong, it can be done without using `_valid_device`:
```python
vol.Required(CONF_DEVICES): {cv.slug: DEVICE_SCHEMA},
```